sphinx_jupyterbook_latex/transforms.py

```diff
@@ -282,3 +285,135 @@ def apply(self, **kwargs: Any) -> None:
             replace_node_cls(bib_node, HiddenCellNode, False)
         if bib_nodes:
             self.document.extend(bib_nodes)
+
+
+class LatexToctreeNodeInterpret(SphinxTransform):
+    """
+    Creates a HiddenCellNode for each toctree node, copies its attributes and inserts
+    it in the doctree adjacent to the toctree node, to be used by a post-transform later.
+    """
+
+    default_priority = 999
+
+    def apply(self, **kwargs: Any) -> None:
+        for tocnode in self.document.traverse(toctree_node):
+            if tocnode.attributes["hidden"]:
+                continue
+
+            compoundnode = tocnode.parent
+            compoundparent = tocnode.parent.parent
+
+            # copy all toctree attributes
+            hiddennode = HiddenCellNode()
+            hiddennode.attributes = tocnode.attributes
+            hiddennode.attributes["classes"] = "latex-tableofcontents"
+
+            # inserts hiddennode adjacent to a toctree node
+            for i, elem in enumerate(compoundparent.children):
+                if elem == compoundnode:
+                    insertionindex = i
+            compoundparent.insert(insertionindex, hiddennode)
+
+
+class SwapTableofContents(SphinxPostTransform):
```

Comment:
@AakashGfude wondering if we should label this as:

`ListTableOfContents`

then when `list` is given for `jblatex_show_tocs` this specific `post_transform` is run?

Reply:
This would then allow other post_transforms to be written with different solutions.

Reply:
@mmcky done in https://github.com/executablebooks/sphinx-jupyterbook-latex/pull/64/commits/ba3b77d279ea6dc572d4dcfb818a2a54b2f08f6b